sklearn/linear_model/_coordinate_descent.py

```diff
@@ -1536,12 +1538,23 @@ def fit(self, X, y, sample_weight=None):
             MSE that is finally used to find the best model is the unweighted
             mean over the (weighted) MSEs of each test fold.
 
+        **params : dict, default=None
+            Parameters to be passed to the CV splitter.
+            These are only available if `enable_metadata_routing=True`. See
+            :ref:`Metadata Routing User Guide <metadata_routing>` for more
+            details.
+
+            .. versionadded:: 1.4
+
         Returns
         -------
         self : object
             Returns an instance of fitted model.
         """
-        _raise_for_unsupported_routing(self, "fit", sample_weight=sample_weight)
+        _raise_for_params(params, self, "fit")
+        if sample_weight is not None and not has_fit_parameter(self, "sample_weight"):
+            raise ValueError("Underlying estimator does not support sample weights.")
```

Comment:
this can't happen, can it? if sample weight is not in the signature, it cannot possibly be passed ;)

Reply:
MultiTaskElasticNetCV and MultiTaskLassoCV do not support sample_weight currently. But now as we have added **params to both, technically someone can pass sample_weight and if they do and it is unchecked this will break the code.

Reply:
the thing is, `sample_weight` could in theory be used in the cv splitter. So we'd need to check, if the splitter doesn't use it, and the estimator also doesn't support it, then raise.

Reply:
What happens if the splitter uses it? Then if we allow sample weight to be passed further than the splitter the place where we fit the base/child model will pass sample weight as well and that won't work with these estimators.